src/pages/admin/create-notes.tsx

```diff
@@ -0,0 +1,418 @@
+          console.log({ "Success mutations.createNotes: ": json });
+          this.setState({ showAlert: 'Saved ✅' });
+          return true;
+
+      }).catch((e: any) => { console.log(e) })
+
+      var updateNotes:any = API.graphql({
+          query: mutations.updateNotes,
+          variables: { input: this.state.noteObject },
+          authMode: GRAPHQL_AUTH_MODE.AMAZON_COGNITO_USER_POOLS
+      });
+
+      updateNotes.then((json: any) => {
+          console.log({ "Success mutations.updateNotes: ": json });
+          this.setState({ showAlert: 'Saved ✅' });
+          return true;
+
+      }).catch((e: any) => { console.log(e) })
+
+      return false;
+    }
+  }
+
+  waitForSelection(conditionFunction: () => boolean) {
+
+    const poll = (resolve: any) => {
+      if(conditionFunction()) resolve();
+      else setTimeout(_ => poll(resolve), 500);
+    }
+  
+    return new Promise(poll);
+  }
+
+  async handleEdit() {
+    this.setState({ showEditModal: true });
+    await this.waitForSelection(() => this.state.noteEditObject);
+    const blocksfromHtml = htmlToDraft(this.state.noteEditObject.content);
+    const { contentBlocks, entityMap } = blocksfromHtml;
+    const contentState = ContentState.createFromBlockArray(contentBlocks, entityMap);
+    const date = parse(this.state.noteEditObject.id, "yyyy-MM-dd", new Date())    
+    this.setState({
+      editorState: EditorState.createWithContent(contentState),
+      selectedTags: this.state.noteEditObject.tags,
+      date: date,
+      title: this.state.noteEditObject.title,
+      pdf: this.state.noteEditObject.pdf
+    })
+    this.updateField('title', this.state.noteEditObject.title) // call updateField to set id
+  }
+
+  updateField(field: any, value: any) {
+    let note = this.state.noteObject
+    note[field] = value
+
+    note.id = format(this.state.date, "yyyy-MM-dd")
+    this.setState({ noteObject: note })
+  }
+
+  onChange = (editorState: any) => this.setState({ editorState });
+
+  handlePublishDate = (date: any) => {
+    this.setState({
+      date: date
+    });
+
+    var dayofweek = getDay(date)
+    if (dayofweek !== 0) {
+      this.setState({
+        dateWarning: 'Warning: this date is not a Sunday'
+      })
+    } else {
+      this.setState({
+        dateWarning: ''
+      })
+    }
+  }
+
+  // RENDER FUNCTIONS
+
+  renderEditModal() {
+    return <Modal isOpen={this.state.showEditModal}>
+        <div>Edit existing notes</div>
+        <select onChange={(event:any) => this.setState({ noteEdit: event.target.value})}>
+          <option key="null" value="null">None Selected</option>
+          {this.state.notesList.map((item: any) => {return <option key={item.id} value={item.id}>{item.id}</option>})}
+        </select>
+        <button 
+          onClick={() => this.setState({ 
+            showEditModal: false, 
+            noteEditObject: this.state.notesList.filter((post: any) => post.id === this.state.noteEdit)[0]
```

Comment:
You don't have to specify types for callback functions like this. `filter` is defined using generics, so you will get the types automatically from its signature.